tests/backup/backup_helper.go

```diff
@@ -5434,3 +5434,128 @@ func RestartAllVMsInNamespace(namespace string, waitForCompletion bool) error {
 	}
 	return nil
 }
+
+// UpgradeKubevirt upgrades the kubevirt control plane to the given version
+// If workloadUpgrade is set to true, kubevirt workloads are upgraded to the given version
+func UpgradeKubevirt(versionToUpgrade string, workloadUpgrade bool) error {
+	k8sKubevirt := kubevirt.Instance()
+	k8sCore := core.Instance()
+
+	// Checking current version
+	current, err := k8sKubevirt.GetVersion()
+	if err != nil {
+		return err
+	}
+	log.Infof("Current version is - %s", current)
+
+	// Compare and validate the upgrade path
+	currentKubevirtVersionSemVer, err := version.NewSemver(strings.TrimSpace(strings.ReplaceAll(current, "v", "")))
+	if err != nil {
+		return err
+	}
+	versionToUpgradeSemVer, err := version.NewSemver(strings.TrimSpace(strings.ReplaceAll(versionToUpgrade, "v", "")))
+	if err != nil {
+		return err
+	}
+	if currentKubevirtVersionSemVer.GreaterThanOrEqual(versionToUpgradeSemVer) {
+		return fmt.Errorf("kubevirt cannot be upgraded from [%s] to [%s]", currentKubevirtVersionSemVer.String(), versionToUpgradeSemVer.String())
+	} else {
+		log.InfoD("Upgrade path chosen (%s) ---> (%s)", current, versionToUpgrade)
+	}
+
+	// Generating the manifest URL and applying it to begin upgrade
+	manifestYamlURL := fmt.Sprintf("https://github.com/kubevirt/kubevirt/releases/download/%s/kubevirt-operator.yaml", versionToUpgrade)
+	_, err = kubectlExec([]string{fmt.Sprintf("--kubeconfig=%v", CurrentClusterConfigPath), "apply", "-f", manifestYamlURL})
+	if err != nil {
+		return err
+	}
+
+	// Waiting till the upgrade is complete
+	t := func() (interface{}, bool, error) {
+		upgradedVersion, err := k8sKubevirt.GetVersion()
+		if err != nil {
+			return "", false, err
+		}
+		if upgradedVersion != versionToUpgrade {
+			return "", true, fmt.Errorf("waiting for kubevirt control plane to be upgraded to [%s] but got [%s]", versionToUpgrade, upgradedVersion)
+		}
+		return "", false, nil
+	}
+	_, err = task.DoRetryWithTimeout(t, 10*time.Minute, 30*time.Second)
+	if err != nil {
+		return err
+	}
+
+	// Waiting for all pods in kubevirt namespace to be Ready
+	kubevirtCheck := func() (interface{}, bool, error) {
+		pods, err := k8sCore.GetPods(KubevirtNamespace, make(map[string]string))
+		if err != nil {
+			return "", false, err
+		}
+		allReady := true
+		for _, p := range pods.Items {
```

Comment:
Kubevirt is a roling update, I feel this logic may not be complete. The running pods will be in ready state, it wont give you correct state during the upgrade.
You should check for pods Age to be not older than upgrade time.